src/coreclr/hosts/inc/coreclrhost.h

```diff
@@ -27,7 +27,8 @@ CORECLR_HOSTING_API(coreclr_initialize,
 
 CORECLR_HOSTING_API(coreclr_shutdown,
             void* hostHandle,
-            unsigned int domainId);
+            unsigned int domainId,
```

Comment:
You should not need to create CorHost4 . The new interface can be implemented by CorHost2 just fine. Also, you should not need to create `coreclr_initialize_2` or `coreclr_execute_assembly_2`.  `coreclr_shutdown_2` is the only one needed.